src/gamma.jl

```diff
@@ -692,33 +701,49 @@ function lgamma(z::Complex{Float64})
     else
         shift = Complex(real(shift), imag(shift) + signflips*6.2831853071795864769252842)
     end
-    return lgamma_asymptotic(Complex(x,y)) - shift
+    return loggamma_asymptotic(Complex(x,y)) - shift
 end
-lgamma(z::Complex{T}) where {T<:Union{Integer,Rational}} = lgamma(float(z))
-lgamma(z::Complex{T}) where {T<:Union{Float32,Float16}} = Complex{T}(lgamma(Complex{Float64}(z)))
+loggamma(z::Complex{T}) where {T<:Union{Integer,Rational}} = loggamma(float(z))
+loggamma(z::Complex{T}) where {T<:Union{Float32,Float16}} = Complex{T}(loggamma(Complex{Float64}(z)))
 
-gamma(z::Complex) = exp(lgamma(z))
+gamma(z::Complex) = exp(loggamma(z))
 
 """
     beta(x, y)
 
 Euler integral of the first kind ``\\operatorname{B}(x,y) = \\Gamma(x)\\Gamma(y)/\\Gamma(x+y)``.
 """
-function beta(x::Number, w::Number)
-    yx, sx = lgamma_r(x)
-    yw, sw = lgamma_r(w)
-    yxw, sxw = lgamma_r(x+w)
+function beta end
+
+function beta(x::Real, w::Real)
+    yx, sx = logabsgamma(x)
+    yw, sw = logabsgamma(w)
+    yxw, sxw = logabsgamma(x+w)
     return exp(yx + yw - yxw) * (sx*sw*sxw)
 end
 
+function beta(x::Number, w::Number)
+    yx = loggamma(x)
+    yw = loggamma(w)
+    yxw = loggamma(x+w)
+    return exp(yx + yw - yxw)
+end
+
 """
-    lbeta(x, y)
+    logbeta(x, y)
 
-Natural logarithm of the absolute value of the [`beta`](@ref)
+Natural logarithm of the [`beta`](@ref)
 function ``\\log(|\\operatorname{B}(x,y)|)``.
 """
-lbeta(x::Number, w::Number) = lgamma(x)+lgamma(w)-lgamma(x+w)
+logbeta(x::Number, w::Number) = loggamma(x)+loggamma(w)-loggamma(x+w)
 
+"""
+    logabsbeta(x, y)
+
+Natural logarithm of the absolute value of the [`beta`](@ref)
+function ``\\log(|\\operatorname{B}(x,y)|)``.
+"""
+logabsbeta(x::Real, w::Real) = logabsgamma(x)[1]+logabsgamma(w)[1]-logabsgamma(x+w)[1]
```

Comment:
For consistency with `logabsgamma`, this should return a tuple
```suggestion
function logabsbeta(x::Real, w::Real)
    yx, sx = logabsgamma(x)
    yw, sw = logabsgamma(w)
    yxw, sxw = logabsgamma(x+w)
    (yx + yw - yxw), (sx*sw*sxw)
end
```

Reply:
In a minute!